### PR TITLE
Add new use cases related to groups and sorting

### LINKED ISSUE #364: [PE-D][Tester E] Important use cases are missing from DG
Some important use cases seem to be missing from the DG. 

For example, the editing group names. I think it is a user interaction worthy of documenting because there are many extensions worth mentioning eg person trying to editing a group that does not exist, trying to edit the name to a name that already exists.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: tingxuanp/ped#10

### PR DESCRIPTION
**Pull request by voldiemort**
Fixes #364 

Not too sure if we need extension 4a. for the sorting use case (UC11). Used to demonstrate that the sort option persists but reads a bit clunkily. Please see and comment/suggest changes

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -412,6 +412,186 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 1a. User may not require a list of groups.
+  
+    Use case resumes at step 2.
+
+* 2a. The group with specified name does not exist.
+  * 2a1. CFG shows an error message.
+
+    Use case ends.
+* 3b. The group currently has no members.
```
Comment (seah-minlong) [inline docs/DeveloperGuide.md:479]: Is this a formatting error

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -412,6 +412,186 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    Use case ends.
+
+
+**Use case: UC07 - View members of a group**
+
+**MSS**
+
+1. User requests to <ins>see a list of groups (UC06)</ins>
+2. User requests to see members of a group
+3. CFG lists all the members of the group
```
Comment (seah-minlong) [inline docs/DeveloperGuide.md:465]: Might have missed out the full stops

Comment (seah-minlong) [reply docs/DeveloperGuide.md:465]: Same for some of the other use cases too

**Review by seah-minlong — CHANGES_REQUESTED** [review]
Looks good but I think might need full-stops. The one given in lecture has full-stops. 🤣 




Also, i think #364 mentions editing group name as an important use case. Should we include that too? I do think we captured most "representative use cases" already though.
image

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (voldiemort) [reply docs/DeveloperGuide.md:479]: Nope, it comes out fine in the actual document! Markdown cares more about the indentation than the newline.

Comment (seah-minlong) [reply docs/DeveloperGuide.md]: shesh ok

**Review by seah-minlong — APPROVED** [review]
LGTM!!

**Pull request merged**